src/torque-offset-estimator.cpp

```diff
@@ -147,48 +143,67 @@ namespace dynamicgraph
         }
         else if (sensor_offset_status == INPROGRESS) {
           // Check the current iteration status
-          int _i = encSignals.size();
-
+          int _i = stdVecJointTorqueOffsets.size();
+          
           if (_i < n_iterations) {
-            SEND_MSG("Collecting signals for iteration no:" + _i , MSG_TYPE_DEBUG);
-            const dynamicgraph::Vector& _enc = m_base6d_encodersSIN(iter);
-            const dynamicgraph::Vector& _acc = m_accelerometerSIN(iter);
-            const dynamicgraph::Vector& _gyr = m_gyroscopeSIN(iter);
-            const dynamicgraph::Vector& _tau = m_jointTorquesSIN(iter);
+            SEND_MSG("Collecting signals for iteration no:" + _i , MSG_TYPE_DEBUG_STREAM);
+
+            const Eigen::VectorXd& sot_enc = m_base6d_encodersSIN(iter);
+            const Eigen::VectorXd& IMU_acc = m_accelerometerSIN(iter);
+            const Eigen::VectorXd& IMU_gyr = m_gyroscopeSIN(iter);
+            const Eigen::VectorXd& _tau = m_jointTorquesSIN(iter);
+
+            Eigen::VectorXd _enc(m_model.nq);
+            base_sot_to_urdf(sot_enc.head<6>(), _enc.head<7>());
+            _enc.tail(m_model.njoints) = sot_enc.tail(m_model.njoints);
+
+            //Get the transformation from ff(f) to torso (t) to IMU(i) frame:
+            // fMi = oMf^-1 * fMt * tMi
+            se3::forwardKinematics(m_model,*m_data,_enc);
+            se3::SE3 fMi = m_data->oMi[ffIndex].inverse()*m_data->oMi[torsoIndex]*m_torso_X_imu;
+
+            //Move the IMU signal to the base frame.
+            //angularAcceleration is zero. Intermediate frame acc and velocities are zero
+            const dynamicgraph::Vector _acc = fMi.rotation()*IMU_acc; //Torso Acceleration
+
+            //Deal with gravity predefined in robot model. Robot Z should be pointing upwards
+            m_model.gravity.linear() = _acc;
+
+            const Eigen::VectorXd& _tau_rnea = se3::rnea(m_model, *m_data, _enc,
+                                                         Eigen::VectorXd::Zero(m_model.njoints),
+                                                         Eigen::VectorXd::Zero(m_model.njoints));
+            const Eigen::VectorXd _current_offset = _tau - _tau_rnea.tail(m_model.njoints);
+            if(_current_offset.array().abs().maxCoeff() >=epsilon) {
+              SEND_MSG("Too high torque offset estimated for iteration"+ _i, MSG_TYPE_ERROR);
+              assert(false);
+            }
+            // encSignals.push_back(_enc);
+            // accSignals.push_back(_acc);
+            // gyrSignals.push_back(_gyr);
+            // tauSignals.push_back(_tau);
+            stdVecJointTorqueOffsets.push_back(_current_offset);
```

Comment:
As I said before, there is no need to keep track of all the values computed at each iteration, just sum them all and at the end you just divide by the number of iterations to get the average.